content/en/blog/2021/upcoming-networking-changes/index.md

```diff
@@ -0,0 +1,71 @@
+
+## Current behavior
+
+In Istio prior to release 1.10, the Envoy proxy, running in the same pod as the application, binds to the `eth0` interface and redirects all inbound traffic to the `lo` interface.
+
+{{< image width="50%" link="./current.svg" caption="A pod's network with Istio today" >}}
+
+This has two important side effects that cause the behavior to differ from standard Kubernetes:
+
+* Applications binding only to `lo` will receive traffic from other pods, when otherwise this is not allowed.
+* Applications binding only to `eth0` will not receive traffic.
+
+Applications that bind to both interfaces (which is typical) will not be impacted.
+
+## Future behavior
+
+Starting with Istio 1.10, the networking behavior is changed to align with the standard behavior present in Kubernetes.
+
+{{< image width="50%" link="./planned.svg" caption="A pod's network with Istio in the future" >}}
+
+Here we can see that the proxy no longer redirects the traffic to the `lo` interface, but instead forwards it to the application on `eth0`.
+As a result, the standard behavior of Kubernetes is retained, but we still get all the benefits of Istio.
+This change allows Istio to get closer to its goal of being a drop-in transparent proxy that works with existing workloads with [zero configuration](/blog/2021/zero-config-istio/).
+Additionally, it avoids unintended exposure of applications binding only to `lo`.
+
+## Am I impacted?
+
+For new users, this change should only be an improvement.
+However, if you are an existing user, you may have come to depend on the old behavior, intentionally or accidentally.
+
+To help detect these situations, we have added a check to find pods that will be impacted.
+You can run the `istioctl experimental precheck` command to get a report of any pods binding to `lo` on a port exposed in a `Service`.
+This command is available in Istio 1.10+.
+Without action, these ports will no longer be accessible upon upgrade.
+
+{{< text bash >}}
+$ istioctl experimental precheck
+Warning [IST0143] (Pod echo-local-849647c5bd-g9wxf.default) Port 443 is exposed in a Service but listens on localhost. It will not be exposed to other pods.
+Warning [IST0143] (Pod echo-local-849647c5bd-g9wxf.default) Port 7070 is exposed in a Service but listens on localhost. It will not be exposed to other pods.
+Error: Issues found when checking the cluster. Istio may not be safe to install or upgrade.
+See https://istio.io/latest/docs/reference/config/analysis for more information about causes and resolutions.
+{{< /text >}}
+
+### Migration
+
+If you are currently binding to `lo`, you have a few options:
+
+* Switch your application to bind to all interfaces (`0.0.0.0` or `::`).
+* Explicitly configure the port using the [`Sidecar` ingress configuration](/docs/reference/config/networking/sidecar/#IstioIngressListener) to send to `lo`, preserving the old behavior.
```

Comment:
Is the configuration for the `port` or `defaultEndpoint` field in sidecar resource?  A simple example would be really helpful here.

Walking through a migration scenario, if I create the sidecar resource for my istio 1.10 cp, and also created my app with 1.10's data plane, would that sidecar resource break traffic for my app with 1.9's data plane? (Correct me if I am wrong, I think our sidecar resource is global for a ns regardless of which control plane revision?)

Reply:
> Is the configuration for the port or defaultEndpoint field in sidecar resource

Both are required fields

The steps would be:
* Create sidecar in 1.9. This is scoped to a workload. It has no impact.
* Upgrade to 1.10. Because of sidecar explicitly saying to connect to localhost, there is no change in behavior

Reply:
Yea was trying to see if we could isolate the sidecar only for 1.10.  I don't think we could.   In that case, user should validate sidecar resource has no impact to the app (1.9 based) first before upgrading to 1.10.

Note: some of the stateful [apps](https://istio.io/latest/faq/applications/), we ask users to change certain configs to listen on localhost, probably a good time to revert that change after testing.

Reply:
You can with istio.io/rev, but there is no real reason to. I already changed that in https://preliminary.istio.io/latest/about/faq#applications.